src/app/pluginmanager/qgspluginmanager.cpp

```diff
@@ -405,121 +405,135 @@ void QgsPluginManager::getCppPluginsMetadata()
+        mPlugins.insert( baseName, metadata );
 
-      QgsDebugMsgLevel( "Loaded library: " + myLib->fileName(), 2 );
-      //Type is only used in non-provider plugins, so data providers are not picked
-      if ( !myLib->resolve( "type" ) )
-      {
         delete myLib;
-        continue;
-      }
-
-      // resolve the metadata from plugin
-      name_t *pName = ( name_t * ) cast_to_fptr( myLib->resolve( "name" ) );
-      description_t *pDesc = ( description_t * ) cast_to_fptr( myLib->resolve( "description" ) );
-      category_t *pCat = ( category_t * ) cast_to_fptr( myLib->resolve( "category" ) );
-      version_t *pVersion = ( version_t * ) cast_to_fptr( myLib->resolve( "version" ) );
-      icon_t *pIcon = ( icon_t * ) cast_to_fptr( myLib->resolve( "icon" ) );
-      experimental_t *pExperimental = ( experimental_t * ) cast_to_fptr( myLib->resolve( "experimental" ) );
-      create_date_t *pCreateDate = ( create_date_t * ) cast_to_fptr( myLib->resolve( "create_date" ) );
-      update_date_t *pUpdateDate = ( update_date_t * ) cast_to_fptr( myLib->resolve( "update_date" ) );
-
-      // show the values (or lack of) for each function
-      if ( pName )
-      {
-        QgsDebugMsgLevel( "Plugin name: " + *pName(), 2 );
-      }
-      else
-      {
-        QgsDebugMsgLevel( QStringLiteral( "Plugin name not returned when queried" ), 2 );
       }
-      if ( pDesc )
+      catch ( QgsSettingsException &ex )
       {
-        QgsDebugMsgLevel( "Plugin description: " + *pDesc(), 2 );
-      }
-      else
-      {
-        QgsDebugMsgLevel( QStringLiteral( "Plugin description not returned when queried" ), 2 );
-      }
-      if ( pCat )
-      {
-        QgsDebugMsgLevel( "Plugin category: " + *pCat(), 2 );
-      }
-      else
-      {
-        QgsDebugMsgLevel( QStringLiteral( "Plugin category not returned when queried" ), 2 );
-      }
-      if ( pVersion )
-      {
-        QgsDebugMsgLevel( "Plugin version: " + *pVersion(), 2 );
-      }
-      else
-      {
-        QgsDebugMsgLevel( QStringLiteral( "Plugin version not returned when queried" ), 2 );
-      }
-      if ( pIcon )
-      {
-        QgsDebugMsgLevel( "Plugin icon: " + *pIcon(), 2 );
-      }
-      else
-      {
-        QgsDebugMsgLevel( QStringLiteral( "Plugin icon not returned when queried" ), 2 );
-      }
-      if ( pCreateDate )
-      {
-        QgsDebugMsgLevel( "Plugin create date: " + *pCreateDate(), 2 );
-      }
-      else
-      {
-        QgsDebugMsgLevel( QStringLiteral( "Plugin create date not returned when queried" ), 2 );
-      }
-      if ( pUpdateDate )
-      {
-        QgsDebugMsgLevel( "Plugin update date: " + *pUpdateDate(), 2 );
-      }
-      else
-      {
-        QgsDebugMsgLevel( QStringLiteral( "Plugin update date not returned when queried" ), 2 );
+        QgsDebugMsg( QStringLiteral( "Unhandled settings exception loading %1: %2" ).arg( lib, ex.what() ) );
+        continue;
```

Comment:
@nyalldawson , is there a need to `delete myLib;` here?

Reply:
I've moved it to a unique_ptr now